packages/block-library/src/gallery/v1/convert-gallery-modal.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import { Button, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LINK_DESTINATION_ATTACHMENT,
+	LINK_DESTINATION_NONE,
+	LINK_DESTINATION_MEDIA,
+} from './constants';
+
+export const convertGallery = ( {
+	clientId,
+	getBlock,
+	replaceBlocks,
+} ) => () => {
+	let link;
+	const {
+		attributes: { sizeSlug, linkTo, images, caption },
+	} = getBlock( clientId );
+
+	switch ( linkTo ) {
+		case 'post':
+			link = LINK_DESTINATION_ATTACHMENT;
+			break;
+		case 'file':
+			link = LINK_DESTINATION_MEDIA;
+			break;
+		default:
+			link = LINK_DESTINATION_NONE;
+			break;
+	}
+	const innerBlocks = images.map( ( image ) =>
+		createBlock( 'core/image', {
+			id: parseInt( image.id, 10 ),
+			url: image.url,
+			alt: image.alt,
+			caption: image.caption,
+			linkDestination: link,
+		} )
+	);
+
+	replaceBlocks(
+		clientId,
+		createBlock(
+			'core/gallery',
+			{ sizeSlug, linkTo: link, caption },
+			innerBlocks
+		)
+	);
+};
+
+export default function ConvertGalleryModal( { onClose, clientId } ) {
+	const { getBlock } = useSelect( blockEditorStore );
+	const { replaceBlocks } = useDispatch( blockEditorStore );
+	return (
+		<Modal
+			closeLabel={ __( 'Close' ) }
+			onRequestClose={ onClose }
+			title={ __( 'Convert to new gallery format' ) }
+			className={ 'wp-block-convert-gallery-modal' }
+			aria={ {
+				describedby: 'wp-block-convert-gallery-modal__description',
+			} }
+		>
+			<p id={ 'wp-block-convert-gallery-modal__description' }>
+				{ __(
+					'You can convert this gallery block to a new format which provides more flexibility, like adding custom links, or custom styles, to individual images.'
+				) }
+			</p>
+			<p id={ 'wp-block-convert-gallery-modal__description' }>
```

Comment:
Duplicate id attribute. Should this second one be removed, or if we need the `describedby` to include both paragraphs, we could add a wrapper element?